src/nsrt_learning/option_learning.py

```diff
@@ -297,24 +311,30 @@ def __init__(self, name: str, operator: STRIPSOperator,
     def _precondition_based_initiable(self, state: State, memory: Dict,
                                       objects: Sequence[Object],
                                       params: Array) -> bool:
-        # The memory here is used to store the absolute params, based on
-        # the relative params and the object states.
-        memory["params"] = params  # store for sanity checking in policy
-        changing_objects = [objects[i] for i in self._changing_parameter_idxs]
-        memory["absolute_params"] = state.vec(changing_objects) + params
+        if self._parameterized:
+            # The memory here is used to store the absolute params, based on
+            # the relative params and the object states.
+            memory["params"] = params  # store for sanity checking in policy
+            changing_objects = [
+                objects[i] for i in self._changing_parameter_idxs
+            ]
+            memory["absolute_params"] = state.vec(changing_objects) + params
         # Check if initiable based on preconditions.
         grounded_op = self._operator.ground(tuple(objects))
         return all(pre.holds(state) for pre in grounded_op.preconditions)
 
     def _regressor_based_policy(self, state: State, memory: Dict,
                                 objects: Sequence[Object],
                                 params: Array) -> Action:
-        # Compute the updated relative goal.
-        assert np.allclose(params, memory["params"])
-        changing_objects = [objects[i] for i in self._changing_parameter_idxs]
-        relative_goal_vec = memory["absolute_params"] - state.vec(
-            changing_objects)
-        x = np.hstack(([1.0], state.vec(objects), relative_goal_vec))
+        if self._parameterized:
+            # Compute the updated relative goal.
+            assert np.allclose(params, memory["params"])
+            changing_objs = [objects[i] for i in self._changing_parameter_idxs]
+            relative_goal_vec = memory["absolute_params"] - state.vec(
+                changing_objs)
+            x = np.hstack(([1.0], state.vec(objects), relative_goal_vec))
```

Comment:
could we instead pull out this line, and in the else clause just do `relative_goal_vec = []` to make it more clear that that's the only difference?